### ISSUE TITLE
Update to later Ktor

### ISSUE THREAD

**Issue by dalewking** [icerockdev/moko-network#200]
I don't have a specific need for it at the moment, but this library is quite a bit behind the latest version of KTOR

**Comment by Alex009** [comment]
as i know ktor not broke backward compatibility. so you can use latest ktor with latest moko-network.
or you got some errors?

**Comment by dalewking** [comment]
I don't get build errors, but does not work when running. I'll try to get something more concrete

**Comment by dalewking** [comment]
My apologies. You are correct and the error was on my side. All I knew was that when I upgraded to latest Ktor my tests quit working.

The problem was with the way that I was generating a random URL for use in the tests. I was using UrlBuilder from Ktor. Before upgrading to latest moko network and Ktor 2.0.3 Url builder accepted an `encodedPath` parameter which I initialized with "/". Ktor 2.0.3 replaced that with a pathSegments parameter, passing into it `listOf("/")` to match the previous behavior. That worked fine with Ktor 2.0.3 but later versions of Ktor made sure that special characters got URL escapes. That made my base URL look like `https://mycompany.com/%2F`. Initializing pathSegments to an empty list fixed the problem.
